core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java

```diff
@@ -57,36 +57,39 @@ public String name() {
   public void initialize(
       AmsClient client, CatalogMeta meta, Map<String, String> properties) {
     this.meta = meta;
-    meta.putToCatalogProperties(org.apache.iceberg.CatalogUtil.ICEBERG_CATALOG_TYPE,
+    meta.putToCatalogProperties(
+        org.apache.iceberg.CatalogUtil.ICEBERG_CATALOG_TYPE,
         meta.getCatalogType());
     tableMetaStore = CatalogUtil.buildMetaStore(meta);
     if (meta.getCatalogProperties().containsKey(CatalogProperties.CATALOG_IMPL)) {
       meta.getCatalogProperties().remove("type");
     }
-    icebergCatalog = org.apache.iceberg.CatalogUtil.buildIcebergCatalog(name(),
-        meta.getCatalogProperties(), tableMetaStore.getConfiguration());
+    icebergCatalog = tableMetaStore.doAs(() -> org.apache.iceberg.CatalogUtil.buildIcebergCatalog(name(),
+        meta.getCatalogProperties(), tableMetaStore.getConfiguration()));
   }
 
   @Override
   public List<String> listDatabases() {
     if (icebergCatalog instanceof SupportsNamespaces) {
-      return tableMetaStore.doAs(() -> ((SupportsNamespaces)icebergCatalog).listNamespaces(Namespace.empty()).stream()
+      return tableMetaStore.doAs(() -> ((SupportsNamespaces) icebergCatalog).listNamespaces(Namespace.empty()).stream()
           .map(namespace -> namespace.level(0)).distinct().collect(Collectors.toList()));
     } else {
-      throw new UnsupportedOperationException(String.format("Iceberg catalog: %s do now implement SupportsNamespaces",
-       icebergCatalog.getClass().getName()));
+      throw new UnsupportedOperationException(String.format(
+          "Iceberg catalog: %s do now implement SupportsNamespaces",
```

Comment:
`"Iceberg catalog: %s do now implement SupportsNamespaces",`
do now  --> does not